opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -435,8 +450,13 @@ public static void addObservation(Event e, JSONObject jsonObject) {
                         }
                     }
                     if (!optionValues.isEmpty()) {
-                        // For options without concepts combine the values into one observation
-                        createObservation(e, jsonObject, optionKeyVals, optionValues);
+                        if (CONCEPT.equals(entity) && shouldBeCombined) {
+                            e.addObs(new Obs(CONCEPT, AllConstants.CHECK_BOX, jsonObject.optString(OPENMRS_ENTITY_ID), jsonObject.optString(OPENMRS_ENTITY_PARENT), optionEntityIds, optionValues, null,
+                                    jsonObject.optString(KEY)));
```

Comment:
Isn't createObservation being called on line 445?

Reply:
Or are we going for duplicate obs?

Reply:
these are combined values hence, only one obs need to be created, This check `shouldBeCombined` ensures that its working only when combined `AllConstants.COMBINE_CHECKBOX_OPTION_VALUES` is added to the field

Reply:
I think you can define, `jsonObject.optString(OPENMRS_ENTITY_ID), jsonObject.optString(OPENMRS_ENTITY_PARENT), jsonObject.optString(KEY)` once, since they're used elsewhere in this block.

Reply:
cool, changing